data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/contingency/CsaProfileContingencyCreator.java

```diff
@@ -0,0 +1,151 @@
+            String equipmentId = contingencyEquipmentPropertyBag.getId(CsaProfileConstants.REQUEST_CONTINGENCIES_EQUIPMENT_ID);
+            String contingentStatus = contingencyEquipmentPropertyBag.get(CsaProfileConstants.REQUEST_CONTINGENCIES_CONTINGENT_STATUS);
+
+            if (!CsaProfileConstants.IMPORTED_CONTINGENT_STATUS.equals(contingentStatus)) {
+                isIncorrectContingentStatus = true;
+                incorrectContingentStatusElements = incorrectContingentStatusElements.concat(equipmentId + " ");
+            } else {
+                String networkElementId = this.getNetworkElementIdInNetwork(equipmentId);
+                if (networkElementId == null) {
+                    isMissingNetworkElement = true;
+                    missingNetworkElements = missingNetworkElements.concat(equipmentId + " ");
+                } else {
+                    contingencyAdder.withNetworkElement(networkElementId);
+                }
+            }
+        }
+
+        if (isIncorrectContingentStatus) {
+            csaProfileContingencyCreationContexts.add(CsaProfileContingencyCreationContext.notImported(contingencyId, contingencyId, contingencyName, ImportStatus.INCONSISTENCY_IN_DATA, "incorrect contingent status for equipments : " + incorrectContingentStatusElements));
+            return;
+        }
+
+        if (isMissingNetworkElement) {
+            csaProfileContingencyCreationContexts.add(CsaProfileContingencyCreationContext.notImported(contingencyId, contingencyId, contingencyName, ImportStatus.INCONSISTENCY_IN_DATA, "missing contingency equipments in network : " + missingNetworkElements));
+            return;
+        }
+
+        contingencyAdder
+                .add();
+        csaProfileContingencyCreationContexts.add(CsaProfileContingencyCreationContext.imported(contingencyId, contingencyId, contingencyName, "", false));
+    }
+
+    private String getNetworkElementIdInNetwork(String networkElementIdInCrac) {
+        String networkElementId = null;
+        Identifiable<?> networkElement = network.getIdentifiable(networkElementIdInCrac);
+        if (networkElement == null) {
+            CgmesBranchHelper cgmesBranchHelper = new CgmesBranchHelper(networkElementIdInCrac, network);
+            if (cgmesBranchHelper.isValid()) {
+                networkElementId = cgmesBranchHelper.getIdInNetwork();
+            }
+        } else {
+            networkElementId = networkElement.getId();
+        }
+        return networkElementId;
+    }
+
+    private boolean isValidInterval(String startTime, String endTime) {
+        OffsetDateTime cracCreationTimeStamp = cracCreationContext.getTimeStamp();
+        OffsetDateTime startDateTime = OffsetDateTime.parse(startTime);
+        OffsetDateTime endDateTime = OffsetDateTime.parse(endTime);
+        return cracCreationTimeStamp.isAfter(startDateTime) && cracCreationTimeStamp.isBefore(endDateTime);
```

Comment:
if cracCreationTimeStamp==startDateTime, or cracCreationTimeStamp==endDateTime, then the interval is valid too.
So i suggest you use : 
```suggestion
        return !cracCreationTimeStamp.isBefore(startDateTime) && !cracCreationTimeStamp.isAfter(endDateTime);
```